float32[] Xsrc
float32[] Ysrc
---
float32[] Xdst
float32[] Ydst